### PR TITLE
Warn before disconnect

### PR DESCRIPTION
**Pull request by danry25**
Seems to work pretty decently, give it a go and gimme some complaints!
warn-before-disconnect

### CODE REVIEW

File: static/js/wicd.js [hunk 1]
```diff
@@ -106,14 +106,37 @@ function scan_networks() {
...
 
-function connect_to_network() {
-  var network_id = $(this).data('id');
-  showAlert('Connecting...', 'info');
-  $.getJSON( "connect/" + network_id, function(data) {
-    showAlert('Please wait (' + data.data + ')', 'info');
-    showAlert('Connecting to ' + data.data, 'info');
-    watch_connection_status();
-  } );
+function network_status_precheck(){
```
Comment (thefinn93) [inline static/js/wicd.js:109]: You literally quadroupled the size of this function

File: static/js/wicd.js [hunk 2]
```diff
@@ -106,14 +106,37 @@ function scan_networks() {
...
+}
+
+function connect_to_network(network_id) {
+  return function(current) {
+    console.log("Preparing to connect to network...", network_id, current);
+    if(current.data.network === undefined) {
+      showAlert('Connecting...', 'info');
+      $.getJSON("connect/" + network_id, function(data) {
+        showAlert('Please wait (' + data.data + ')', 'info');
+        showAlert('Connecting to ' + data.data, 'info');
```
Comment (thefinn93) [inline static/js/wicd.js]: Wait so you're updating the alert twice in a row? Won't that first one just burn CPU cycles?

File: static/js/wicd.js [hunk 3]
```diff
@@ -106,14 +106,37 @@ function scan_networks() {
...
+        watch_connection_status();
+      });
+    } else {
+      $("#confirmDisconnectText").text('Are you sure you want to disconnect from ' + current.data.network);
+      $("#confirmDisconnectModal").modal("show");
+      $('#confirmDisconnectModal').submit(function(event) {
+        $(".disconnectNetworkButton").addClass("disabled");
```
Comment (thefinn93) [inline static/js/wicd.js]: You should update the text of the button to be like "connecting" or something

File: static/js/wicd.js [hunk 4]
```diff
@@ -106,14 +106,37 @@ function scan_networks() {
...
+        event.preventDefault();
```
Comment (thefinn93) [inline static/js/wicd.js]: Due to reasons, this should be the first call in the function (move it up a line).

File: static/js/wicd.js [hunk 5]
```diff
@@ -106,14 +106,37 @@ function scan_networks() {
...
+        showAlert('Connecting...', 'info');
+        $.getJSON("connect/" + network_id, function(data) {
+          showAlert('Please wait (' + data.data + ')', 'info');
+          showAlert('Connecting to ' + data.data, 'info');
```
Comment (thefinn93) [inline static/js/wicd.js]: Again with the duplicate alert showing

**Review by thefinn93 — CHANGES_REQUESTED** [review]
This looks pretty good! There are a couple of minor inefficiencies that I highlighted, and one or two shitposts. go do the thing

File: static/js/wicd.js (hunk 1 above)
Comment (danry25) [reply static/js/wicd.js:109]: *quadrupled, might I suggest a [linter for your prose](https://atom.io/packages/linter-write-good)? :D

File: static/js/wicd.js (hunk 2 above)
Comment (danry25) [reply static/js/wicd.js]: Axed lines 119 and 130, as they were never displayed.

File: static/js/wicd.js (hunk 3 above)
Comment (danry25) [reply static/js/wicd.js]: Yeah, that is a good idea, I've added that.

File: static/js/wicd.js (hunk 4 above)
Comment (danry25) [reply static/js/wicd.js]: Fixed, I see this is commonly at the top in other parts of the JS, probably does something important I should Google.

File: static/js/wicd.js (hunk 5 above)
Comment (danry25) [reply static/js/wicd.js]: Already got rid of this, thanks tho.

File: static/js/wicd.js [hunk 6]
```diff
@@ -124,6 +124,7 @@ function connect_to_network(network_id) {
       $("#confirmDisconnectModal").modal("show");
       $('#confirmDisconnectModal').submit(function(event) {
         $(".disconnectNetworkButton").addClass("disabled");
+        $(".disconnectNetworkButton").text('Connecting...');
```
Comment (thefinn93) [inline static/js/wicd.js]: whats with the random indent here?

**Review by thefinn93 — APPROVED** [review]

**Pull request merged**